src/main/java/hudson/remoting/RemoteInvocationHandler.java

```diff
@@ -118,26 +120,60 @@
     private final Throwable origin;
 
     /**
+     * Indicates that the handler operates in the user space.
+     * In such case the requests will be automatically failed if the
+     * Remoting channel is not fully operational.
+     */
+    private final boolean userSpace;
+
+    /**
      * Creates a proxy that wraps an existing OID on the remote.
      */
-    RemoteInvocationHandler(Channel channel, int id, boolean userProxy, boolean autoUnexportByCaller, Class proxyType) {
+    RemoteInvocationHandler(Channel channel, int id, boolean userProxy,
+                            boolean autoUnexportByCaller, boolean userSpace,
+                            Class proxyType) {
         this.channel = channel == null ? null : channel.ref();
         this.oid = id;
         this.userProxy = userProxy;
         this.origin = new Exception("Proxy "+toString()+" was created for "+proxyType);
         this.autoUnexportByCaller = autoUnexportByCaller;
+        this.userSpace = userSpace;
+    }
+
+
+
+    /**
+     * Wraps an OID to the user-spaced typed wrapper.
+     *
+     * RPC operations with this object will be checking the channel state before sending the request.
+     * @see UserRPCRequest
+     */
+    @Nonnull
+    public static <T> T wrapUser(Channel channel, int id, Class<T> type, boolean userProxy, boolean autoUnexportByCaller) {
+        return internalWrap(channel, id, type, userProxy, autoUnexportByCaller, true);
+    }
+
+    /**
+     * Wraps an OID to the system-spaced typed wrapper.
+     *
+     * This method is restricted to internal {@link Channel}-specific use only.
+     * Usage of this method in external logic or on the {@link VirtualChannel} level is not supported.
+     */
+    @Nonnull
+    public static <T> T wrapSystem(Channel channel, int id, Class<T> type, boolean userProxy, boolean autoUnexportByCaller) {
+        return internalWrap(channel, id, type, userProxy, autoUnexportByCaller, false);
     }
 
     /**
      * Wraps an OID to the typed wrapper.
      */
     @Nonnull
-    public static <T> T wrap(Channel channel, int id, Class<T> type, boolean userProxy, boolean autoUnexportByCaller) {
+    private static <T> T internalWrap(Channel channel, int id, Class<T> type, boolean userProxy, boolean autoUnexportByCaller, boolean userSpace) {
```

Comment:
For use from `Channel.export` it would be simpler to just make this the only method, and delete the `wrapSystem` and `wrapUser` convenience methods.